File: app/src/main/java/com/bth/running/dagger/AppComponent.java
package com.bth.running.dagger;

import com.bth.running.fragments.CoachFragment;
import com.bth.running.fragments.HistoryFragment;
import com.bth.running.fragments.RunningFragment;

import javax.inject.Singleton;

import dagger.Component;

/**
 * @author Martin Macheiner
 *         Date: 05.09.2017.
 */

@Singleton
@Component(modules = {AppModule.class, NetworkModule.class})
public interface AppComponent {

    void inject(RunningFragment fragment);

    void inject(HistoryFragment fragment);

    void inject(CoachFragment fragment);

}

File: app/src/main/java/at/shockbytes/corey/running/dagger/AppComponent.kt
package at.shockbytes.corey.running.dagger


import at.shockbytes.corey.running.core.LocationRunningService
import at.shockbytes.corey.running.ui.fragment.CoachFragment
import at.shockbytes.corey.running.ui.fragment.HistoryFragment
import at.shockbytes.corey.running.ui.fragment.RunningFragment
import dagger.Component
import javax.inject.Singleton

/**
 * @author  Martin Macheiner
 * Date:    05.09.2017.
 */

@Singleton
@Component(modules = [(AppModule::class), (NetworkModule::class)])
interface AppComponent {

    fun inject(fragment: RunningFragment)

    fun inject(fragment: HistoryFragment)

    fun inject(fragment: CoachFragment)

    fun inject(service: LocationRunningService)

}
